server/src/services/poaching-incident.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable, RequestTimeoutException } from '@nestjs/common';
+import { DatabaseService } from './db.service';
+import { PoachingIncident } from '../entity/poaching-incident.entity';
+import { PoachingIncidentType } from "../entity/poaching-incident-type.entity";
+
+@Injectable()
+export class PoachingIncidentService {
+  constructor(private readonly databaseService: DatabaseService) { }
+
+  async addPoachingIncident(long: number, lat: number, pType: string): Promise<boolean> {
+    const con = await this.databaseService.getConnection();
+    const poachingIncident = new PoachingIncident();
+
+    try {
+      const poachingIncidentType = await JSON.parse(
```

Comment:
No need to do await on synchronous operations like JSON.parse/JSON.stringify. Also there shouldn't be a need to do JSON.parse(JSON.stringify(...)) here. Should work like this I believe:

```
const poachingIncidentType = await con.getRepository(PoachingIncidentType).findOne({ type: pType });
```

Reply:
Okay will keep in mind for future